### DISCUSSION TITLE
empty macro define lead to compile klee fail？

### DISCUSSION THREAD

**Discussion by llmagic** [UnitTestBot/UTBotCpp#278] [Q&A] [answered]
generate tests failed and report `“INTERNAL：Couldn't compile klee file from correct methods”`
In log, detal information shows below 
![image](https://user-images.githubusercontent.com/6960007/172549313-9727f57b-25cc-4beb-9a44-ffcdb4d692ab.png)
The macro EXPORTED is defined in cmakelists.txt, which is mean to define it empty
![image](https://user-images.githubusercontent.com/6960007/172549650-ff1fed18-6ccd-4165-9c0f-7e1af461ac46.png)
and in compile_commands.json it is defined as
`-DEXPORTED=\"\"
`
when I change it to `-DEXPORTED=` in compile_commands.json, compile could pass.
Is that an bug here for compiling klee file?

**Comment by alexey-utkin** [comment, marked as answer]
@llmagic Thank you for your report! 
It seems that the problem comes from the `bear` utility.  The `bear` is used to capture the compiler setup. We are trying to localize and fix the problem, you can monitor the progress in the [issue ](https://github.com/UnitTestBot/UTBotCpp/issues/276).
As work around you can manually correct the lines like this
```
       "command": "/utbot_distr/install/bin/clang++ -DEXPORTS=\"\"  -std=gnu++11 -o CMakeFiles/example1.dir/main.cpp.o -c /home/utbot/tmp/dev/example1/main.cpp",
```
from `compile_commands.json` to 
```
       "command": "/utbot_distr/install/bin/clang++ -DEXPORTS=  -std=gnu++11 -o CMakeFiles/example1.dir/main.cpp.o -c /home/utbot/tmp/dev/example1/main.cpp",
```
